### ISSUE TITLE
[TASK] Add network selection component and implement network-based page access in Tangle dApp

### ISSUE THREAD

**Issue by devpavan04** [webb-tools/webb-dapp#2089]
### Overview

Currently, the `tangle-dapp` connects to the `Tangle Testnet` by default. However, we should enable users to connect to various networks, including:

- Tangle Mainnet
- Tangle Testnet
- Custom Endpoint

To implement this, consider using the network selection component from the [stats-dapp](https://stats.tangle.tools/#/keys) as a reference. This component allows users to input custom endpoints and connect to either the Tangle `Mainnet` or `Testnet`.

![CleanShot 2024-03-12 at 11 44 29](https://github.com/webb-tools/webb-dapp/assets/53374218/b526d3b4-ec7f-437a-9d1e-40508b4aba34)

![CleanShot 2024-03-12 at 11 44 34](https://github.com/webb-tools/webb-dapp/assets/53374218/c3ab3ec2-c577-4549-963a-e68979dda76a)

Additionally, we should implement network-based access restrictions for certain pages on the `tangle-dapp`, depending on the user's selected network. For example, the Restaking feature should be available on the Testnet but not on the Mainnet for now. This restriction can be enforced by displaying a `Not Supported` modal.

**Comment by monaiuu** [comment]
### Checklist

- [ ] **Add Network Switch Component:** using the network selection component from the [stats-dapp](https://stats.tangle.tools/#/keys)  to allow users to input custom endpoints and connect to either the Tangle Mainnet or Testnet.
- [ ]  **Implement Non-Supported Modal:** Create a user-friendly modal to inform users when a feature is not available on their current network.
- [ ] **Identify Features/Pages for Gating:** List features/pages that should be gated on specific networks.

**Comment by drewstone** [comment]
The UI here in the issue should be updated CC @monaiuu

**Comment by monaiuu** [comment]
## Updated Handoff
Update network switching feature, allowing users to seamlessly toggle between mainnet, testnet, and a custom RPC endpoint.

Reference: For an example of a similar implementation, see the network switch component at [app.webb.tools](https://app.webb.tools/).

### Custom Endpoint Flow
![Image](https://github.com/webb-tools/webb-dapp/assets/109539270/2b6dc9b1-ede4-4eb8-89de-ef65e5d7d75d)


- Step 1: The user interacts with the network switch component, which displays three options: Mainnet, Testnet, and Development.
Upon selecting "Development," a dropdow input field is triggered for the user to input their RPC URL.

- Step 2: The user enters the RPC endpoint URL into the input field and submits it for validation.

- Step 3: Feedback and Error Handling
  - If Valid: Update UI to reflect the switch to `Custom Endpoint,` and proceed to switch the network connection to use the provided custom RPC endpoint.
  - If Invalid or Unresponsive: Provide the user with an error message indicating the failure to connect or validate the RPC endpoint.

**Comment by monaiuu** [comment]
@yurixander I also updated the above comment to completely hide `Service Overview` and `Restake` pages from the side navigation so it is not accessible for public for launch. Only if connected to development would these pages be accessible.

**Comment by yurixander** [comment]
> Gate Service Overview & Restake: Remove Service Overview Page & Restake Page from side navigation. Only if user is in development would this be accessible.

Will be handling this on a separate issue.
